03_letters/starfallz/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, strings.Join(sortLetters(letters("aba")), "\n"))
+}
+
+func letters(s string) map[rune]int {
+	letterCount := make(map[rune]int)
+
+	for _, currentLetter := range s {
+		letterCount[currentLetter]++
+	}
+
+	return letterCount
+}
+
+func sortLetters(m map[rune]int) []string {
+	allKeys := make([]rune, 0, len(m))
+	for k := range m {
+		allKeys = append(allKeys, k)
+	}
+
+	sortRune(allKeys)
+
+	result := make([]string, len(allKeys))
+
+	for i, currentKey := range allKeys {
+		result[i] = strings.Join([]string{string(currentKey), strconv.Itoa(m[currentKey])}, ":")
+	}
```

Comment:
As a suggestion I'd try to do it without creating an intermediate slice - that's how I did it, but not yet reviewed, so not sure if it's the right way to go but I do think so. Makes things a lot shorter too.